Add "How to Document Adapter" guide (#1467)
fishtown-analytics/docs.getdbt.com: website/docs/docs/contributing/documenting-a-new-adapter.md

---
title: "Documenting a new adapter"
id: "documenting-a-new-adapter"
---

If you've already [built an adapter](/website/docs/docs/contributing/building-a-new-adapter.md), and [tested your adapter]((/website/docs/docs/contributing/testing-a-new-adapter.md)), it's time to document it so the dbt community will know that it exists and how to use it!

## General Guidelines

### Single Source of Truth

> We have asked our partner adapter maintainers to use the [docs.getdbt.com repo](https://github.com/dbt-labs/docs.getdbt.com) (i.e. this site) as the single-source-of-truth for documentation rather than having to maintain the same set of information in three different places. Instead, we ask that the adapter repo's `README.md` and the dataplatform's documentation pages simply link to the corresponding page on this docs site.

Reviewer: ```suggestion
We asked our adapter maintainers to use the [docs.getdbt.com repo](https://github.com/dbt-labs/docs.getdbt.com) (i.e. this site) as the single-source-of-truth for documentation rather than having to maintain the same set of information in three different places. The adapter repo's `README.md` and the data platform's documentation pages should simply link to the corresponding page on this docs site.
```

Cleaning this up so it's more direct. Also should we be linking to the repo? I think it makes sense but also sounds a bit odd on the docs site that is created from the repo.

|----------------------|--------------------------------------------------------------|--------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Connect              | `reference/warehouse-profiles/{MY-DATA-PLATOFRM}-profile.md` | Create | Give all information needed to define a target in `~/.dbt/profiles.yml` and get `dbt debug` to connect to the database successfully. All possible configurations should be mentioned.                |
| Configure            | `reference/resource-configs/{MY-DATA-PLATOFRM}-configs.md`   | Create | What options and configuration specific to your data platform do users need to know? e.g. table distribution and indexing options, column_quoting policy, which incremental strategies are supported |
| Discover and Install | `docs/available-adapters.md`                                 | Modify | Is adapter vendor- or community- supported? How to install Python adapter package? Ideally with pip and PyPI hosted package, but can also use `git+` link to GitHub Repo                             |

> The following three pages need to be modified or created to have your data platform be listed on our documentation. After the corresponding pull request is merged, we ask that you link to these pages from your adapter repo's `REAMDE` as well as from your product documentation

Reviewer: Could we move this to be above the table? Reading this quickly, it's not clear to me that I need to create or modified these new packages specifically in the folders in the repo. So kinda more words to be lear that you are going to be creating say a new profile page for just your data platform in the `warehouse-profiles` folder.

Author: done! lmk what you think now

## General Guidelines
To best inform the dbt community of the new adapter, you should contribute to the dbt's open-source documentation site, which uses the [Docusauraus project](https://docusaurus.io/). This is the site you're currently on!

### Conventions

Each `.md` file you create needs a header as shown below. The document id will also need to be added to the config file: `website/sidebars.js`.

```
---
title: "Documenting a new adapter"
id: "documenting-a-new-adapter"
---
```

### Single Source of Truth

We ask our adapter maintainers to use the [docs.getdbt.com repo](https://github.com/dbt-labs/docs.getdbt.com) (i.e. this site) as the single-source-of-truth for documentation rather than having to maintain the same set of information in three different places. The adapter repo's `README.md` and the data platform's documentation pages should simply link to the corresponding page on this docs site. Keep reading for more information on what should and shouldn't be included on the dbt docs site.

### Assumed Knowledge

> To simplify things, assume the reader of this documentation already knows how both dbt and your data platform works. There's already great material for how to learn dbt and the data platform out there. The documentation we're asking you to add should be what a user who is already profiecient in both dbt and your data platform would need to know in order to use both. Effectively that boils down to two things: how to connect, and how to configure.

Reviewer: Yesss this is so much better

---
title: "Documenting a new adapter"
id: "documenting-a-new-adapter"
---

> If you've already [built an adapter](/website/docs/docs/contributing/building-a-new-adapter.md), and [tested your adapter]((/website/docs/docs/contributing/testing-a-new-adapter.md)), it's time to document it so the dbt community will know that it exists and how to use it!

Reviewer: ```suggestion
If you've already [built](/website/docs/docs/contributing/building-a-new-adapter.md), and [tested]((/website/docs/docs/contributing/testing-a-new-adapter.md)) your adapter, it's time to document it so the dbt community will know that it exists and how to use it!
```

## General Guidelines

> To best inform the dbt community of the new adapter, you will be contribtute to dbt's open-source documentation site, which uses the [Docusauraus project](https://docusaurus.io/).

Reviewer: ```suggestion
To best inform the dbt community of the new adapter, you should contribute to the dbt's open-source documentation site, which uses the [Docusauraus project](https://docusaurus.io/). This is the site you're currently on!
```

### Conventions

> Each `.md` file you create needs a header that looks like the below. The document id given below will also be need to be added to the config file: `website/sidebars.js`.

Reviewer: ```suggestion
Each `.md` file you create needs a header as shown below. The document id will also need to be added to the config file: `website/sidebars.js`.
```

```
---
title: "Documenting a new adapter"
id: "documenting-a-new-adapter"
---
```

### Single Source of Truth

> We asked our adapter maintainers to use the [docs.getdbt.com repo](https://github.com/dbt-labs/docs.getdbt.com) (i.e. this site) as the single-source-of-truth for documentation rather than having to maintain the same set of information in three different places. The adapter repo's `README.md` and the data platform's documentation pages should simply link to the corresponding page on this docs site.

Reviewer: I feel like we need to offer more guidance on what belongs on the data platform's docs site vs ours.  Like installation and dbt configurations should go on our site but like how the adapter works with any of their product features might make sure on theirs (or something like that)

Author: i added more info. the sections below should address your concern now?

### Assumed Knowledge

> To simplify things, assume the reader of this documentation already knows how both dbt and your data platform works. There's already great material for how to learn dbt and ostensibly for the adapter's back end data platform.

Reviewer: ```suggestion
To simplify things, assume the reader of this documentation already knows how both dbt and your data platform works. There's already great material for how to learn dbt and the data platform out there.
```

Just making this sentence a little more clear

## Topics and Pages to Cover

The following subjects need to be addressed across three pages of this docs site to have your data platform be listed on our documentation. After the corresponding pull request is merged, we ask that you link to these pages from your adapter repo's `REAMDE` as well as from your product documentation.

>  To contribute all you will have to do make the changes listed in the table below.

Reviewer: ```suggestion
 To contribute, all you will have to do make the changes listed in the table below.
```

>  For example say I want to document my new adapter: `dbt-ders`. For the "Connect" page, I will make a new Markdown file, `ders-profile.md` and add it to the `website/docs/reference/warehouse-profiles/` directory.

Reviewer: Could this come after the table?

Author: done!

| How To...            | File to change within `/website/docs/`                       | Action | Info to Include                                                                                                                                                                                      |
|----------------------|--------------------------------------------------------------|--------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Connect              | `reference/warehouse-profiles/{MY-DATA-PLATOFRM}-profile.md` | Create | Give all information needed to define a target in `~/.dbt/profiles.yml` and get `dbt debug` to connect to the database successfully. All possible configurations should be mentioned.                |
| Configure            | `reference/resource-configs/{MY-DATA-PLATOFRM}-configs.md`   | Create | What options and configuration specific to your data platform do users need to know? e.g. table distribution and indexing options, column_quoting policy, which incremental strategies are supported |

> | Discover and Install | `docs/available-adapters.md`                                 | Modify | Is adapter vendor- or community- supported? How to install Python adapter package? Ideally with pip and PyPI hosted package, but can also use `git+` link to GitHub Repo                             |

Reviewer: ```suggestion
| Discover and Install | `docs/available-adapters.md`                                 | Modify | Is it a vendor- or community- supported adapter? How to install Python adapter package? Ideally with pip and PyPI hosted package, but can also use `git+` link to GitHub Repo                             |
```
